src/main/groovy/lang/ObjectRange.java

```diff
@@ -339,24 +335,23 @@ public List subList(int fromIndex, int toIndex) {
 
         // Performance detail:
         // not using get(fromIndex), get(toIndex) in the following to avoid stepping over elements twice
-        final StepIterator iter = new StepIterator(this, 1);
```

Comment:
I measure the difference using JMH on a single thread:

```java
final static List<Integer> iterable = List.range(1, 1000000);

public static class Iter {
    @Benchmark
    public int synchronized_iteration() {
        final Iterator<Integer> it = new Iterator<Integer>() {
            private final Iterator<Integer> delegate = iterable.iterator();

            @Override
            public synchronized boolean hasNext() {
                return delegate.hasNext();
            }

            @Override
            public synchronized Integer next() {
                return delegate.next();
            }
        };

        int result = 0;
        while (it.hasNext()) {
            result ^= it.next();
        }
        return result;
    }

    @Benchmark
    public int normal_iteration() {
        final Iterator<Integer> it = new Iterator<Integer>() {
            private final Iterator<Integer> delegate = iterable.iterator();

            @Override
            public boolean hasNext() {
                return delegate.hasNext();
            }

            @Override
            public Integer next() {
                return delegate.next();
            }
        };

        int result = 0;
        while (it.hasNext()) {
            result ^= it.next();
        }
        return result;
    }
}
```
resulting in:
```
Impl                    Score    ±  Error  Unit   normal_iteration  synchronized_iteration
normal_iteration        193.828  ±  0.82%  ops/s                               1.01x
synchronized_iteration  192.702  ±  0.44%  ops/s       0.99x                              
```
i.e. same speed in a non-contended environment (the synchronized is probably JITed away)